server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/ParsedSignificantTerms.java

```diff
@@ -175,7 +175,7 @@ public final XContentBuilder toXContent(XContentBuilder builder, Params params)
                         bucket.subsetDf = value;
                         bucket.setDocCount(value);
                     } else if (InternalSignificantTerms.SCORE.equals(currentFieldName)) {
-                        bucket.score = parser.longValue();
+                        bucket.score = parser.doubleValue();
```

Comment:
Are there any examples of existing unit tests for "toXContent -> fromXContent" serialization on aggregation buckets I can base this on?

Reply:
as far as I can see BaseAggregationTestCase has a testFromXContent method. So SignficantTermsTests sounds like the appropriate test.

Reply:
>BaseAggregationTestCase has a testFromXContent method
In whic case we might be missing some test infrastructure for aggs here. That methods for testing agg requests but we have no equivalent for testing responses. I guess `toXcontent->fromXContent` is a transformation of response objects that didn't exist before high level rest client.

Reply:
Right, that is for requests, my bad, but we have tests for responses too, for instance `InternalTermsTestCase`

Reply:
Had a quick chat with @colings86 and it looks like we need a base class for aggs with an abstract `assertParsedResponse(InternalX, ParsedX)`. InternalX here is the object used to hold the response on the server and ParsedX is the client-side equivalent.

That work's beyond the scope of this fix so I suggest we open another issue to address that broader set of work and put this PR to bed as-is?

Reply:
Good catch, @colings86. The buckets need `updateScore` calling on them in the test code to derive scores from all the frequency stats. Otherwise tests were always testing for score=0 conditions and failing to highlight the bug.
When I remove `fromXContent` fix and run with the beefed up tests it now reproduces the original error so it looks like the test infra is there.

Reply:
glad to hear,  am a bit rusty on this topic but I kind of knew/hoped that we have added tests when adding parsing for the high-level REST client.